oidc_provider/lib/endpoints/token.py

```diff
@@ -299,6 +326,14 @@ def create_client_credentials_response_dic(self):
 
         token.save()
 
+        token_created.send(
```

Comment:
Could this be put somewhere else instead of being duplicated?

Reply:
I don't think we gain a lot by just moving this elsewhere as most of the arguments are different each time it is called. I'm not entirely sure where to shove these are there each currently placed where they have access to all the information they need to send in that case. They could be combined with the save method to emphasize those actions should happen together? @adamlynam-catalyst what do you think?

Reply:
Based on my understanding of the code I think it would be hard to remove the duplication without simply adding another copy of it elsewhere with some common variables.

I think this is probably close to the best we can do already.